candle-transformers/src/models/metavoice.rs

```diff
@@ -0,0 +1,316 @@
+use candle::{DType, IndexOp, Module, Result, Tensor, D};
+use candle_nn::{linear_no_bias, Linear, VarBuilder};
+
+pub struct SpeakerEncoder;
+pub struct BPETokenizer;
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash)]
+pub enum NormType {
+    LayerNorm,
+    RMSNorm,
+}
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash)]
+pub enum AttnKernelType {
+    Fa2,
+    TorchAttn,
+    Hand,
+}
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash)]
+pub enum NonLinearityType {
+    Gelu,
+    Swiglu,
+}
+
+enum Norm {
+    RMSNorm(candle_nn::RmsNorm),
+    LayerNorm(candle_nn::LayerNorm),
+}
+
+fn linear(in_dim: usize, out_dim: usize, bias: bool, vb: VarBuilder) -> Result<Linear> {
+    if bias {
+        candle_nn::linear(in_dim, out_dim, vb)
+    } else {
+        linear_no_bias(in_dim, out_dim, vb)
+    }
+}
+
+// https://github.com/metavoiceio/metavoice-src/blob/11550bb4e8a1ad032cc1556cc924f7a4e767cbfa/fam/llm/model.py#L27
+#[derive(Debug, Clone)]
+pub struct GPTConfig {
+    pub block_size: usize,
+    pub vocab_sizes: Vec<usize>,
+    pub n_layer: usize,
+    pub n_head: usize,
+    pub n_embd: usize,
+    pub bias: bool,
+    pub causal: bool,
+    pub spk_emb_on_text: bool,
+    pub norm_type: NormType,
+    pub rmsnorm_eps: f64,
+    pub nonlinearity_type: NonLinearityType,
```

Comment:
nit: this is legacy part of our code, we don't use this anymore! Everything released should be SwiGLU only